addons/csw/functions/fnc_reload_handleAddTurretMag.sqf

```diff
@@ -41,12 +47,18 @@ if (_canAdd) then {
         TRACE_2("Setting mag ammo",_loadedMag,_currentAmmo);
         // _vehicle setMagazineTurretAmmo [_loadedMag, _currentAmmo, _turret];
 
-        // setMagazineTurretAmmo is broken on split locality, use setAmmo for now (this may not work for multi turret vehicles)
+        // setMagazineTurretAmmo is broken on split locality and if there are multiple magazines of the same class (valid 20/12/2023, Arma 2.14)
+        // use setAmmo for now (this may not work for multi turret vehicles)
         private _weapon = (_vehicle weaponsTurret _turret) param [0, ""];
         TRACE_3("setAmmo",_vehicle,_weapon, _currentAmmo);
         _vehicle setAmmo [_weapon, _currentAmmo];
         private _currentAmmo = _vehicle magazineTurretAmmo [_loadedMag, _turret];
-        if ((_weapon == "") || {_currentAmmo != _currentAmmo}) then { ERROR_1("failed to setAmmo - %1", _this); };
+        if ((_weapon == "") || {_currentAmmo != _currentAmmo}) then {
+            // Don't remove ammo if it failed to load
+            _ammoRemaining = _ammoRemaining + _ammoUsed;
+
+            ERROR_1("failed to setAmmo - %1",_this);
```

Comment:
I've adding this change because I feel it's wrong to punish a player if a reload can't happen, but I'm not sure if this can be abused.